src/jukebox/run_jukebox.py

```diff
@@ -1,19 +1,121 @@
+#!/usr/bin/env python3
+"""
+Top-level entry point for the JukeBox Core demon
+
+##############
+Configuration
+##############
+TBD
+
+##############
+Logger
+##############
+We use a hierarchical Logger structure based on pythons logging module. It can be finely configured with a yaml file.
+
+The top-level logger is called 'jb' (to make it short). In any module you may simple create a child-logger at any hierarchy
+level below 'jb'. It will inherit settings from it's parent logger unless otherwise configured in the yaml file.
+Hierarchy separator is the '.'. If the logger already exits, getLogger will return a reference to the same, else it will be
+created on the spot.
+
+:Example::
+    >>> import logging
+    >>> logger = logging.getLogger('jb.awesome_module')
+    >>> logger.info('Started general awesomeness aura')
+
+Example: YAML snippet, setting WARNING as default level everywhere and DEBUG for jb.awesome_module::
+``
+loggers:
+  jb:
+    level: WARNING
+    handlers: [console, debug_file_handler, error_file_handler]
+    propagate: no
+  jb.awesome_module:
+    level: DEBUG
+``
+
+.. note::
+The name (and hierarchy path) of the logger can be arbitrary and must not necessarily match the module name (still makes sense)
+There can be multiple loggers per module, e.g. for special classes, to further control the amount of log output
+"""
+
 import os.path
+import sys
 import argparse
+import logging
+import logging.config
+from ruamel.yaml import YAML
 import jukebox.daemon
+import misc.loggingext
 
 
-if __name__ == "__main__":
+def logger_default(level=logging.DEBUG):
```

Comment:
Could `logger_default` and below `logger_configure` be its own module which we can import here? we could also move the nice readme above to this file and keep the `run_jukebox.py` clean

Reply:
agree, in addition to the readme for user, we should created a kind development handbook where those very valuable information should be placed.
I don't think someone would necessarily read this source file, when adding e.g. a new components module.

Reply:
Done: moved everything into misc.loggingext.py. 
This just calls for proposing Sphinx + RST documentation. You get to write the documentation in the Python source code files and later have nice documentation. It's a bit of work to set it up though...